src/relay/pushservice/pushservice.py

```diff
@@ -51,21 +50,33 @@ def dedup_event_id(client_token, event: Event):
         return None
 
 
+def create_firebase_app_from_path_to_keyfile(path_to_keyfile: str):
+    cred = credentials.Certificate(path_to_keyfile)
+
+    app = firebase_admin.initialize_app(cred)
+
+    return app
+
+
 class FirebaseRawPushService:
     """Sends push notifications to firebase. Sending is done based on raw client tokens"""
 
-    def __init__(self, path_to_keyfile: str) -> None:
+    def __init__(self, app) -> None:
         """
         Initializes the push service
         Args:
-            path_to_keyfile: Path to json keyfile with firebase credentials
+            app: The initialized firebase_admin App
         """
-        cred = credentials.Certificate(path_to_keyfile)
-        self._app = firebase_admin.initialize_app(cred)
+
+        if not isinstance(app, firebase_admin.App):
```

Comment:
I would not do such a strict check, as it destroys the flexibility of python to just pass in something wich provides the right methods.